### PR TITLE
Remote global address book updates

### LINKED ISSUE #5625: Provide remote updates for the global address book
## Abstract

We have a global name registry (address book) in the global DB. This is editable by the user. But unlike other global data such as assets, spam, assets, rpc nodes etc. we do not push data updates for it yet.

## Task

The approach should be same as in #5532 

Related issues: 

- #5594 
- #5532 
- #5236

### PR DESCRIPTION
**Pull request by nebolax**
Closes #5625

## Checklist

- [ ] The PR modified the frontend, and updated the [user guide](https://github.com/rotki/rotki/blob/develop/docs/usage_guide.rst) to reflect the changes.

### CODE REVIEW

File: rotkehlchen/tests/unit/test_data_updates.py [hunk 1]
```diff
@@ -172,3 +208,41 @@ def test_update_rpc_nodes(data_updater: RotkiDataUpdater) -> None:
...
+    db_addressbook = DBAddressbook(data_updater.user_db)
+    # check state of the address book before updating
+    with GlobalDBHandler().conn.read_ctx() as cursor:
+        cursor.execute('SELECT COUNT(*) FROM address_book')
+        assert cursor.fetchone()[0] == 0
+
+    # Add some initial entries. "Local name 1" doesn't conflict with any of the remote names
+    # and should stay as is after the update application. "Local name 2" conflicts with
+    # "Remote name 3", but since it is local, it should be preferred over the remote entry
+    # and should also stay in the db as is.
```
Comment (prettyirrelevant) [inline rotkehlchen/tests/unit/test_data_updates.py:224]:
```suggestion
    # populate the `addressbook` table and see that existing local entries are preferred to their remote counterpart
```

**Review by prettyirrelevant — COMMENTED** [review]
A question.

If an update contains a wrong name for an address, and a new update is created(containing the right name for that entry), your current implementation won't handle that right?

Should that be an expected behaviour?

**Comment by nebolax** [comment]
> A question.
> If an update contains a wrong name for an address, and a new update is created(containing the right name for that entry), your current implementation won't handle that right?
> Should that be an expected behaviour?

I would say that it is kinda overkill~ish. I would ask @LefterisJP whether this is needed.

**Comment by LefterisJP** [comment]
> > A question.
> > If an update contains a wrong name for an address, and a new update is created(containing the right name for that entry), your current implementation won't handle that right?
> > Should that be an expected behaviour?
> 
> I would say that it is kinda overkill~ish. I would ask @LefterisJP whether this is needed.

We should be able to update names. That's an important point of the existence of remote data. Say if an address is compromised. First it's a Kraken address. Then something happens and it's a hacked compromised address. We would update our remote data and this should be pulled downstream.

File: rotkehlchen/db/updates.py [hunk 2]
```diff
@@ -154,6 +162,47 @@ def update_rpc_nodes(self, to_version: int) -> None:
...
+                    ),
+                )
+                if existing_name is None:
+                    entries_to_add.append(entry)
+                else:
+                    log.debug(
+                        f'Global address book entry for address {entry.address} and blockchain '
+                        f'{entry.blockchain} already exists (probably was input by the user). '
+                        f'Update for this entry will not be applied.',
+                    )
```
[the remark this answers is not in the document]
Comment (nebolax) [reply rotkehlchen/db/updates.py:198]: In the other PR (curve api pr) Lefteris said that we should not add things like `exists_ok` flag to `add_addresbook_entries` method, so I did it like this here. I'll also add the check here whether to force insert remote name, and for this it will probably be needed to get current local names.

**Review by prettyirrelevant — COMMENTED** [review]
some comments

File: docs/changelog.rst [hunk 3]
```diff
@@ -3,6 +3,7 @@ Changelog
 =========
 
 * :feature:`5816` The NFT images will not be automatically rendered now. It is made so to prevent a known security issue, that may result in leakage of your privacy (read https://medium.com/@alxlpsc/critical-privacy-vulnerability-getting-exposed-by-metamask-693c63c2ce94 ). You can add domains you trust to the whitelisted domain in the NFT setting.
+* :feature:`5625` Global address book will now receive automatic updates regularly i.e. known addresses will be associated with names.
```
Comment (LefterisJP) [inline docs/changelog.rst:6]: Perhaps let's not advertise this. As discussed this will only have some of our small updates to our own remote data and not the usage of other sources. So such a changelog probably may create expectations

File: rotkehlchen/db/updates.py [hunk 4]
```diff
@@ -154,6 +163,70 @@ def update_rpc_nodes(self, to_version: int) -> None:
...
+                existing_name = db_addressbook.get_addressbook_entry_name(
+                    book_type=AddressbookType.GLOBAL,
+                    chain_address=OptionalChainAddress(
+                        address=address,
+                        blockchain=blockchain,
+                    ),
+                )
+                if existing_name is None:
+                    entries_to_add.append(entry)
+                elif raw_entry['old_name'] is not None and existing_name == raw_entry['old_name']:
```
Comment (LefterisJP) [inline rotkehlchen/db/updates.py:206]: If for some reason we miss it you were gonna get a key error here

```suggestion
                elif raw_entry.get('old_name') is not None and existing_name == raw_entry['old_name']:
```

File: rotkehlchen/tests/unit/test_data_updates.py [hunk 5]
```diff
@@ -5,21 +5,71 @@
...
+        address=ADDRESSBOOK_ADDRESS_2,
+        blockchain=SupportedBlockchain.ETHEREUM,
+    ),
+]
+SERIALIZED_REMOTE_ADDRESSBOOK = [
+    {
+        'address': ADDRESSBOOK_ADDRESS_1,
+        'blockchain': 'ETH',
+        'new_name': 'Remote name that doesnt conflict with anything',
+        'old_name': None,
```
Comment (LefterisJP) [inline rotkehlchen/tests/unit/test_data_updates.py:48]: I would advise putting `old_name` as optional. A lot less data to serialize for only an edge case. Only add `old_name` in the remote if it's needed. It's easy to guard against this in the app code as I showed you in the other comment and this really helps us as makes the size of the data in the remote and that needs to be serialized over the network smaller.

**Review by LefterisJP — COMMENTED** [review]
This needs a rebase and conflicts to be solved.

Also some more comments to act on

**Review by LefterisJP — APPROVED** [review]
LGTM
